File: solutions/2026-09-19/java/2026-09-19-circle-and-rectangle-overlapping.java
```java
/**
 * Checks if a circle and an axis-aligned rectangle overlap.
 * The circle is defined by its radius and center coordinates (xCenter, yCenter).
 * The rectangle is defined by its bottom-left (x1, y1) and top-right (x2, y2) corners.
 *
 * Link: https://leetcode.com/problems/circle-and-rectangle-overlapping/
 *
 * Approach:
 * The core idea is to find the closest point on the rectangle to the center of the circle.
 * If the distance from the circle's center to this closest point is less than or equal to the circle's radius,
 * then there is an overlap.
 *
 * To find the closest point:
 * 1. For the x-coordinate:
 *    - If xCenter is less than x1, the closest x on the rectangle is x1.
 *    - If xCenter is greater than x2, the closest x on the rectangle is x2.
 *    - Otherwise (xCenter is between x1 and x2, inclusive), the closest x on the rectangle is xCenter itself.
 * 2. Similarly, for the y-coordinate:
 *    - If yCenter is less than y1, the closest y on the rectangle is y1.
 *    - If yCenter is greater than y2, the closest y on the rectangle is y2.
 *    - Otherwise (yCenter is between y1 and y2, inclusive), the closest y on the rectangle is yCenter itself.
 *
 * Let (closestX, closestY) be this closest point.
 * The squared distance between the circle's center (xCenter, yCenter) and (closestX, closestY) is
 * (xCenter - closestX)^2 + (yCenter - closestY)^2.
 *
 * The circle and rectangle overlap if this squared distance is less than or equal to radius^2.
 * We use squared distances to avoid using Math.sqrt(), which can be computationally more expensive and
 * might introduce floating-point precision issues.
 *
 * Time Complexity: O(1) - The calculations are constant time, regardless of input size.
 * Space Complexity: O(1) - Only a few variables are used, constant extra space.
 */
class Solution {
    public boolean checkOverlap(int radius, int xCenter, int yCenter, int x1, int y1, int x2, int y2) {
        // Find the closest x-coordinate on the rectangle to the circle's center.
        // If xCenter is to the left of the rectangle, the closest x is x1.
        // If xCenter is to the right of the rectangle, the closest x is x2.
        // Otherwise, the closest x is xCenter itself (meaning it's within the rectangle's x-bounds).
        int closestX = Math.max(x1, Math.min(xCenter, x2));

        // Find the closest y-coordinate on the rectangle to the circle's center.
        // If yCenter is below the rectangle, the closest y is y1.
        // If yCenter is above the rectangle, the closest y is y2.
        // Otherwise, the closest y is yCenter itself (meaning it's within the rectangle's y-bounds).
        int closestY = Math.max(y1, Math.min(yCenter, y2));

        // Calculate the distance between the circle's center and the closest point on the rectangle.
        // We use squared distance to avoid using Math.sqrt() and potential floating-point inaccuracies.
        int distanceX = xCenter - closestX;
        int distanceY = yCenter - closestY;

        // The squared distance is distanceX^2 + distanceY^2.
        // The circle and rectangle overlap if this squared distance is less than or equal to the squared radius.
        return (distanceX * distanceX) + (distanceY * distanceY) <= (radius * radius);
    }
}
```

File: solutions/2026-09-19/python/2026-09-19-circle-and-rectangle-overlapping.py
```python
# Problem: Circle and Rectangle Overlapping
# Link: https://leetcode.com/problems/circle-and-rectangle-overlapping/
# Approach:
# The core idea is to find the closest point on the rectangle to the center of the circle.
# If the distance between this closest point and the circle's center is less than or equal to the circle's radius,
# then there is an overlap.
#
# To find the closest point on the rectangle to the circle's center (xCenter, yCenter):
# 1. Clamp xCenter to be within the rectangle's x-bounds [x1, x2]. Let this be closestX.
#    If xCenter < x1, closestX = x1.
#    If xCenter > x2, closestX = x2.
#    Otherwise, closestX = xCenter.
# 2. Clamp yCenter to be within the rectangle's y-bounds [y1, y2]. Let this be closestY.
#    If yCenter < y1, closestY = y1.
#    If yCenter > y2, closestY = y2.
#    Otherwise, closestY = yCenter.
#
# The closest point on the rectangle to the circle's center is (closestX, closestY).
#
# Then, calculate the squared distance between (xCenter, yCenter) and (closestX, closestY).
# This squared distance is (xCenter - closestX)^2 + (yCenter - closestY)^2.
#
# The circle and rectangle overlap if this squared distance is less than or equal to the squared radius (radius^2).
# Using squared distances avoids the need for square root operations, which are computationally more expensive.
#
# Time Complexity: O(1) - All operations are constant time.
# Space Complexity: O(1) - Constant extra space is used.

class Solution:
    def checkOverlap(self, radius: int, xCenter: int, yCenter: int, x1: int, y1: int, x2: int, y2: int) -> bool:
        """
        Checks if a circle and an axis-aligned rectangle overlap.

        Args:
            radius: The radius of the circle.
            xCenter: The x-coordinate of the circle's center.
            yCenter: The y-coordinate of the circle's center.
            x1: The x-coordinate of the bottom-left corner of the rectangle.
            y1: The y-coordinate of the bottom-left corner of the rectangle.
            x2: The x-coordinate of the top-right corner of the rectangle.
            y2: The y-coordinate of the top-right corner of the rectangle.

        Returns:
            True if the circle and rectangle overlap, False otherwise.
        """

        # Find the closest x-coordinate on the rectangle to the circle's center.
        # This is done by clamping xCenter to the range [x1, x2].
        closestX = max(x1, min(xCenter, x2))

        # Find the closest y-coordinate on the rectangle to the circle's center.
        # This is done by clamping yCenter to the range [y1, y2].
        closestY = max(y1, min(yCenter, y2))

        # Calculate the squared distance between the circle's center and the closest point on the rectangle.
        # We use squared distance to avoid using sqrt, which is computationally more expensive.
        distanceSquared = (xCenter - closestX)**2 + (yCenter - closestY)**2

        # The circle and rectangle overlap if the squared distance is less than or equal to the squared radius.
        return distanceSquared <= radius**2

```